include/maidsafe/crux/detail/multiplexer.hpp

```diff
@@ -387,8 +389,9 @@ inline void multiplexer::do_start_receive()
 inline void multiplexer::discard_message() {
     boost::system::error_code error;
     endpoint_type remote_endpoint;
+    static char empty_buffer_storage;
     next_layer().receive_from
-        ( boost::asio::buffer(static_cast<char*>(nullptr), 0)
+        ( boost::asio::buffer(&empty_buffer_storage, 1)
```

Comment:
Niall, could we have a protected function (e.g. peek_storage_buffer()) which would contain the static char storage and return boost::asio::buffer? We could then easily use a buffer of size 0 on Linux and Windows and of size 1 on FreeBSD